projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/utils/block-content.ts

```diff
@@ -84,7 +84,10 @@ export function getTextContentFromSelectedBlocks(): GetTextContentFromBlocksProp
 		count: blocks.length,
 		clientIds,
 		content: blocks
-			? blocks.map( block => getBlockTextContent( block.clientId ) ).join( HTML_JOIN_CHARACTERS )
+			? blocks
+					.filter( block => block !== null && block !== undefined ) // Safeguard against null or undefined blocks
```

Comment:
Should we make the condition stronger, checking, for instance, the name object?
```suggestion
					.filter( block => !! block?.name ) // Safeguard against null or undefined blocks
```

Reply:
`block !== null && block !== undefined` is the same as `block != null`